% Chapter Template

\chapter{Background} % Main chapter title

\label{Chapter2} % Change X to a consecutive number; for referencing this chapter elsewhere, use \ref{ChapterX}

\lhead{Chapter 2. \emph{Background}} % Change X to a consecutive number; this is for the header on each page - perhaps a shortened title

This chapter introduces the concepts and techniques that are relevant throughout this thesis. First, the concept of similarity search, especially the two software suite FASTA and BLAST would open our chapter. And then, we would introduce various contemporary concepts in bioinformatics and bioinformatics-related infrastructure such as UniProt and MEDLINE. Additionally, we would introduce the concept of named entity recognition (NER) within the field of Natural Language Processing and how it would be relevant for us. Finally we would see how our project relates to previous works in similar topics and how it would improve, provide alternative or give additional insight to them.

%----------------------------------------------------------------------------------------
%	SECTION 1
%----------------------------------------------------------------------------------------

\section{FASTA and BLAST}

As the title of this thesis already conveyed, the main idea of this project is to bridge the accessibility and knowledge gap between sequence and the main source of knowledge and reference of previous discoveries -- a vast corpora of publications in natural sciences -- through a modern search engine. Given a sequence of amino acids, it would be impossible for a human to directly identify directly the protein, let alone the characteristics and the functions and the characteristics of the protein.

Several attempts on bridging one component of the gap, specifically between sequence and other known sequences, was done in eighties and earlier nineties. In 1981, Smith and Walterman published the algorithm computing complete local sequence alignment, which was further improved by Gotoh in 1982 \citep{gotoh1982improved} and Altschul (Altschul and Erickson, 1986 \citep{altschul1986optimal}). This was however deemed too slow, especially if used for the purpose of one-against-all search, which was heavily (and still is) used for sequence-based knowledge discovery in biomedical research.

In 1985, Lipman and Pearson published the first paper mentioning the DNA and protein sequence alignment program FASTA \citep{Lipman85}. During the first publication, FASTA was designed and intended to search for similar protein sequences. It takes a sequence of amino acids and searches against entries within a corresponding database by using local sequence alignment to find similar sequences. In general, FASTA takes four steps in computing three scores that characterize sequence similarity \citep{Pearson19905}:

\begin{enumerate}
\item Finding identify regions with high density of sequence identities and pair identities between two sequences. FASTA achieved a fast computation in this step by using a look up table, a map that describes for each character where it appears within sequence. In conjunction with the lookup table, FASTA also uses the diagonal method to find 
all regions of similarity between the two sequences, counting matches and penalizing for intervening mismatches. This diagonal could be visually seen in two sequence alignment as series of matches ('dots') in match matrix between two sequences.

\item Rescanning of the 10 regions with highest sequence identities using PAM250 matrix. PAM250 matrix refers to assumed point accepted mutation (PAM) matrix after 250 mutations, which is basically the 250-th power of initial PAM matrix. The probability of each entry within PAM matrix was acquired from analysis of phylogenetic trees (Dayhoff, 1978 \citep{Dayhoff1978model}).

\item Annealing of both ends of alignment and calculating similarity score is the sum of the joined initial regions minus a penalty (usually 20) for each gap \citep{Pearson19905}.

\item Construction of optimal alignment using Needleman-Wunsch Algorthm \citep{needleman1970general} on the best matching region. The program would then return the similarity score of this alignment along with the best score from step 2 and 3.
\end{enumerate}

In 1988, Pearson and Lipman improved the software by adding support and improvement, among others, for nucleic acid similarity search, translated nucleic acid search \citep{PearsonLipman88}. This allowed researchers to do trans-domain search between nucleic and amino acids.

Further down the road, in 1990, Altschul et al. published the Basic Alignment Research Tool \citep{Altschul90}, better known in its acronym as BLAST. The algorithm, like FASTA, is based on heuristics search and is structured in similar manner to BLAST. BLAST takes a sequence to search for and a sequence or a set of sequences to search against. In modern usage, the set of sequences is provided by some database. The algorithm would then run in following main steps\citep{mount2001bioinformatics}:

\begin{enumerate}
\item Removal of low complexity regions or sequence repeats from query sequence. Low complexity refers to sequence with few elements.
\item Creation of k-gram sequences from query sequence.
\item For each word from step 2, listing of possible matching words and selection of high scoring words. Matching words are the all possible combinations of words with same length as the k-gram word. For each possible word a score is calculated, which is based on substitution matrix. The best scoring words are then passed onto next step. This differs from FASTA, which focuses more on common words in database.
\item Organization of remaining high scoring words into efficient search three. Both step 3 and 4 would be repeated for each word from step 2.
\item Scanning of database for exact matches with remaining high-scoring words.
\item Extension of database match to high-scoring-segment pair (HSP). This is done by annealing both ends of match until the matching score begins to decrease.
\item Listing of all HSPs that are significant enough.
\item Evaluation of statistical significance of the HSPs. BLAST models statistical significance using Gumbel extreme value distribution \citep{gumbel1954statistical}, in which the probability of observing score $S$ higher than equal to $x$ is defined as $$P(S \leq x) = 1 - exp(-e^{- \lambda (x - \mu)})$$ with $$\mu = log(Km'n')/t$$ The parameters $\mu$ and $K$ are fitted from the distribution of results from high scoring pairs. m' and n' are effective length of the query and database sequences.
\item Make two or more HSP regions into one alignment. In a given hit sequence from database, the algorithm would attempt merging the regions into one had the score of combined region is larger than individual score.
\item Computation of sequence alignments using Smith-Walterman Algorithm \citep{smith1981identification}.
\end{enumerate}

Nowadays, both FASTA and BLAST were distributed not only locally but also online by various providers such as National Center of Biotechnology Information (NCBI) \footnote{\href{http://blast.ncbi.nlm.nih.gov/Blast.cgi}{\texttt{http://blast.ncbi.nlm.nih.gov/Blast.cgi}}} and European Bioinformatics Institute (EBI) \footnote{\href{http://www.ebi.ac.uk/Tools/sss/wublast/}{\texttt{http://www.ebi.ac.uk/Tools/sss/wublast/}}} \footnote{\href{http://www.ebi.ac.uk/Tools/sss/fasta/}{\texttt{http://www.ebi.ac.uk/Tools/sss/fasta/}}}.

%----------------------------------------------------------------------------------------
%	SECTION 3
%----------------------------------------------------------------------------------------

\section{Natural Language Processing}

The rapid development in sequence similarity search coupled with explosion of genome-wide sequencing, which was even more augmented by the advent of post-Sanger and -- recently -- New Generation Sequencing (NGS), means that the problem of identifying sequence is more or less explained. There is however one part that is missing from our picture: how to get the information on how the sequence was mentioned in previous publications?

\subsection{Named Entity Recognition}

%----------------------------------------------------------------------------------------
%	SECTION 2
%----------------------------------------------------------------------------------------

\section{Previous Works}

There are several previous works that we are aware of that try to tackle similar problems. 